### PR TITLE
Add formatMultipleRanges to DocumentRangeFormattingFeature

### PR DESCRIPTION
**Pull request by c-claeys**
Add multirange formatting support via new request textDocument/rangesFormatting
---
~This does nothing besides break `npm run compile` ATM, but I'm looking for advice. Am I missing some hook to correctly include the proposed API? I received the proposed .d.ts via `npx vscode-dts dev`, but `compile` doesn't seem to pick it up AFAICT.~

### CODE REVIEW

**Comment by c-claeys** [comment]
Sounds good. I was just looking at past discussions. Initially we were thinking of putting additional ranges in the formatting options but VSCode's API sync resulted in parallel functions exposed on that API. Any opinion over whether the formatting options should be modified to include ranges, if a new formatting method (`textDocument/rangesFormatting`) should be added, or something else?

File: protocol/src/common/protocol.ts [hunk 1]
```diff
@@ -3448,10 +3448,34 @@ export interface DocumentRangeFormattingParams extends WorkDoneProgressParams {
 	options: FormattingOptions;
 }
 
+/**
```
Comment (dbaeumer) [inline protocol/src/common/protocol.ts:3451]: We need a @since tag here and a @proposed tag

File: protocol/src/common/protocol.ts [hunk 2]
```diff
@@ -3448,10 +3448,34 @@ export interface DocumentRangeFormattingParams extends WorkDoneProgressParams {
...
+}
+
 /**
  * Provider options for a {@link DocumentRangeFormattingRequest}.
  */
 export interface DocumentRangeFormattingOptions extends WorkDoneProgressOptions {
+	/**
+     * Whether the server supports formatting multiple ranges at once.
+     */
+	canFormatMultipleRanges?: boolean;
```
Comment (dbaeumer) [inline protocol/src/common/protocol.ts:3478]: to make things consitent with other name and the client capability I would name it `multipleRangesSupport` or `rangesSupport`

**Review by dbaeumer — CHANGES_REQUESTED** [review]

File: protocol/src/common/protocol.ts (hunk 2 above)
Comment (c-claeys) [reply protocol/src/common/protocol.ts]: Done.

File: protocol/src/common/protocol.ts (hunk 1 above)
Comment (c-claeys) [reply protocol/src/common/protocol.ts:3459]: Done.

**Comment by c-claeys** [comment]
> Otherwise (as always) great PR.

Cheers! Thanks for the reviews

**Comment by dbaeumer** [comment]
PR looks good. @c-claeys are you providing a spec change as well?

**Review by dbaeumer — APPROVED** [review]

**Review by jrieken — APPROVED** [review]

**Comment by c-claeys** [comment]
Sure, happy to follow up with a spec change PR

**Pull request merged**